src/ui/MissingTerm/MissingTerm.ts

```diff
@@ -0,0 +1,102 @@
+import 'styling/_MissingTerm';
+import { exportGlobally } from '../../GlobalExports';
+import { Component } from '../Base/Component';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { ComponentOptions } from '../Base/ComponentOptions';
+import { reject } from 'underscore';
+import { $$, Initialization } from '../../Core';
+import { Dom } from '../../utils/Dom';
+import { analyticsActionCauseList, IAnalyticsIncludeMissingTerm } from '../Analytics/AnalyticsActionListMeta';
+
+export interface IMissingTermOptions {
+  clickable?: boolean;
+}
+
+export class MissingTerm extends Component {
+  static ID = 'MissingTerm';
+  static options: IMissingTermOptions = {
+    /**
+     * Enable the user to click on a missing term to force it in the query
+     */
+    clickable: ComponentOptions.buildBooleanOption({ defaultValue: true })
+  };
+
+  static doExport = () => {
+    exportGlobally({
+      MissingTerm: MissingTerm
+    });
+  };
+  /**
+   * The `MissingTerm` component is responsible for displaying the keyword that was not matched for a result
+   */
+  constructor(public element: HTMLElement, public options?: IMissingTermOptions, bindings?: IComponentBindings, public result?) {
+    super(element, MissingTerm.ID, bindings);
+
+    this.options = ComponentOptions.initComponentOptions(element, MissingTerm, options);
+    this.addMissingTerm();
+  }
+
+  private addMissingTerm() {
+    if (this.getMissingTerms().length > 0) {
+      const container = this.buildContainer();
+      $$(this.element).append(container.el);
+      const cell = $$(this.element).find('.coveo-result-cell');
+      const caption = this.buildCaption();
+      cell.appendChild(caption.el);
+      this.buildMissingTerms().forEach(term => {
+        cell.appendChild(term.el);
+      });
+    }
+  }
+
+  private buildContainer(): Dom {
+    return $$('div', { className: 'coveo-result-row' }, $$('div', { className: 'coveo-result-cell' }));
+  }
+
+  public getMissingTerms(): string[] {
```

Comment:
I'd prefer for all the public method to be grouped together at the top of the class, right after the constructor.

Public methods will also need public documentation, because we'd want dev to interact with the component through them.